Clean/Circomlib/BinSub.lean

```diff
@@ -0,0 +1,147 @@
+import Clean.Circuit
+import Clean.Utils.Bits
+import Clean.Gadgets.Bits
+import Clean.Gadgets.Boolean
+
+
+
+namespace Circomlib
+open Utils.Bits
+variable {p : ℕ} [Fact p.Prime] [Fact (p > 2)]
+
+-- Instance to show that 2 > 0
+instance : Fact (0 < 2) := ⟨by norm_num⟩
+
+-- Define a 2D vector type for BinSub inputs
+-- Represents 2 operands, each with n bits
+-- This is a vector of 2 elements, where each element is a vector of n field elements
+@[reducible]
+def BinSubInput (n : ℕ) := ProvableVector (fields n) 2
+
+-- Define output type for BinSub
+-- Represents n output bits
+@[reducible]
+def BinSubOutput (n : ℕ) := fields n
```

Comment:
nit: unnecessary definition, we can just use `fields`

Reply:
removed